app/browser/api/ledger.js

```diff
@@ -569,9 +571,14 @@ const getPublisherData = (result, scorekeeper) => {
     pinPercentage: result.pinPercentage,
     weight: result.pinPercentage
   }
-  // HACK: Protocol is sometimes blank here, so default to http:// so we can
-  // still generate publisherURL.
-  data.publisherURL = (result.protocol || 'http:') + '//' + result.publisherKey
+
+  data.publisherURL = result.publisherURL || ((result.protocol || 'http:') + '//' + result.publisherKey)
+
+  // media publisher
+  if (result.faviconName) {
+    const on = locale.translation('on').toLowerCase()
+    data.siteName = `${result.faviconName} ${on} ${result.providerName}`
```

Comment:
[minor] this might cause the wrong word order in some languages. i think we support l10n parameters to solve this problem, but maybe not in the `locale` module

Reply:
done